src/compiler/factory.ts

```diff
@@ -3490,12 +3506,16 @@ namespace ts {
             if (firstOriginalDeclaration && isBindingPattern(firstOriginalDeclaration.name)) {
                 // This works whether the declaration is a var, let, or const.
                 // It will use rhsIterationValue _a[_i] as the initializer.
-                const declarations = flattenVariableDestructuring(
+                const declarations = flattenDestructuringBinding(
+                    context,
                     firstOriginalDeclaration,
                     elementAccess,
-                    visitor,
-                    /*recordTempVariable*/ undefined,
+                    /*skipInitializer*/ false,
+                    /*recordTempVariablesInLine*/ true,
                     convertObjectRest

```

Comment:
This is not an issue now that I've merged in the work from `isolateObjectSpread`.